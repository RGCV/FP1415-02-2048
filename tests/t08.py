## ========================================================================== ##
## -------------------------------------------------------------------------- ##
##                                  TEST 08                                   ##
## -------------------------------------------------------------------------- ##
## ========================================================================== ##

# -- Modifier: board_reduce -- #
B = create_board()
C = create_coordinate(1, 4)
B = board_fill_position(B, C, 2)
B = board_reduce(B, 'W')
board_score(B)
B = board_fill_position(B, C, 2)
B = board_reduce(B, 'W')
board_score(B)
# - (additional hidden tests) - #
B = create_board()
B = board_fill_position(B, create_coordinate(1, 4), 2)
B = board_fill_position(B, create_coordinate(2, 3), 2)
B = board_fill_position(B, create_coordinate(3, 2), 2)
B = board_fill_position(B, create_coordinate(4, 1), 2)
B = board_reduce(B, 'W')
print_board(B)
B = create_board()
B = board_fill_position(B, create_coordinate(1, 3), 2)
B = board_fill_position(B, create_coordinate(1, 4), 2)
B = board_fill_position(B, create_coordinate(2, 2), 2)
B = board_fill_position(B, create_coordinate(2, 3), 2)
B = board_fill_position(B, create_coordinate(3, 1), 2)
B = board_fill_position(B, create_coordinate(3, 2), 2)
B = board_fill_position(B, create_coordinate(4, 1), 2)
B = board_fill_position(B, create_coordinate(4, 4), 2)
B = board_reduce(B, 'E')
print_board(B)
B = create_board()
B = board_fill_position(B, create_coordinate(1, 2), 2)
B = board_fill_position(B, create_coordinate(1, 4), 2)
B = board_fill_position(B, create_coordinate(2, 1), 2)
B = board_fill_position(B, create_coordinate(2, 3), 2)
B = board_fill_position(B, create_coordinate(3, 1), 2)
B = board_fill_position(B, create_coordinate(3, 2), 2)
B = board_fill_position(B, create_coordinate(3, 3), 2)
B = board_fill_position(B, create_coordinate(4, 1), 2)
B = board_fill_position(B, create_coordinate(4, 3), 2)
B = board_fill_position(B, create_coordinate(4, 4), 2)
B = board_reduce(B, 'W')
print_board(B)
B = create_board()
B = board_fill_position(B, create_coordinate(1, 1), 2)
B = board_fill_position(B, create_coordinate(1, 3), 4)
B = board_fill_position(B, create_coordinate(1, 4), 4)
B = board_fill_position(B, create_coordinate(2, 1), 2)
B = board_fill_position(B, create_coordinate(2, 2), 2)
B = board_fill_position(B, create_coordinate(2, 3), 2)
B = board_fill_position(B, create_coordinate(2, 4), 2)
B = board_fill_position(B, create_coordinate(3, 2), 2)
B = board_fill_position(B, create_coordinate(3, 3), 2)
B = board_fill_position(B, create_coordinate(4, 1), 2)
B = board_fill_position(B, create_coordinate(4, 2), 2)
B = board_fill_position(B, create_coordinate(4, 4), 2)
B = board_reduce(B, 'N')
print_board(B)
B = create_board()
B = board_fill_position(B, create_coordinate(1, 1), 2)
B = board_fill_position(B, create_coordinate(1, 2), 2)
B = board_fill_position(B, create_coordinate(1, 4), 4)
B = board_fill_position(B, create_coordinate(2, 1), 2)
B = board_fill_position(B, create_coordinate(2, 2), 2)
B = board_fill_position(B, create_coordinate(2, 3), 4)
B = board_fill_position(B, create_coordinate(3, 2), 4)
B = board_fill_position(B, create_coordinate(3, 3), 2)
B = board_fill_position(B, create_coordinate(3, 4), 2)
B = board_fill_position(B, create_coordinate(4, 1), 4)
B = board_fill_position(B, create_coordinate(4, 3), 2)
B = board_fill_position(B, create_coordinate(4, 4), 2)
B = board_reduce(B, 'S')
print_board(B)
B = create_board()
B = board_fill_position(B, create_coordinate(1, 1), 2)
B = board_fill_position(B, create_coordinate(1, 3), 2)
B = board_fill_position(B, create_coordinate(1, 4), 4)
B = board_fill_position(B, create_coordinate(2, 2), 2)
B = board_fill_position(B, create_coordinate(2, 3), 2)
B = board_fill_position(B, create_coordinate(2, 4), 4)
B = board_fill_position(B, create_coordinate(3, 1), 4)
B = board_fill_position(B, create_coordinate(3, 2), 2)
B = board_fill_position(B, create_coordinate(4, 1), 4)
B = board_fill_position(B, create_coordinate(4, 3), 2)
B = board_reduce(B, 'W')
print_board(B)
B = create_board()
B = board_fill_position(B, create_coordinate(1, 1), 2)
B = board_fill_position(B, create_coordinate(1, 4), 4)
B = board_fill_position(B, create_coordinate(2, 2), 2)
B = board_fill_position(B, create_coordinate(2, 3), 4)
B = board_fill_position(B, create_coordinate(3, 1), 2)
B = board_fill_position(B, create_coordinate(3, 3), 4)
B = board_fill_position(B, create_coordinate(4, 3), 4)
B = board_fill_position(B, create_coordinate(4, 4), 2)
B = board_reduce(B, 'E')
print_board(B)
B = create_board()
B = board_fill_position(B, create_coordinate(1, 2), 2)
B = board_fill_position(B, create_coordinate(1, 4), 2)
B = board_fill_position(B, create_coordinate(2, 1), 2)
B = board_fill_position(B, create_coordinate(3, 1), 4)
B = board_fill_position(B, create_coordinate(3, 2), 4)
B = board_fill_position(B, create_coordinate(3, 3), 4)
B = board_fill_position(B, create_coordinate(4, 3), 2)
B = board_fill_position(B, create_coordinate(4, 4), 4)
B = board_reduce(B, 'N')
print_board(B)
B = create_board()
B = board_fill_position(B, create_coordinate(1, 1), 4)
B = board_fill_position(B, create_coordinate(1, 2), 4)
B = board_fill_position(B, create_coordinate(1, 4), 2)
B = board_fill_position(B, create_coordinate(2, 2), 2)
B = board_fill_position(B, create_coordinate(2, 3), 2)
B = board_fill_position(B, create_coordinate(3, 1), 2)
B = board_fill_position(B, create_coordinate(3, 3), 4)
B = board_fill_position(B, create_coordinate(3, 4), 8)
B = board_fill_position(B, create_coordinate(4, 3), 8)
B = board_fill_position(B, create_coordinate(4, 4), 8)
B = board_reduce(B, 'S')
print_board(B)
B = create_board()
B = board_fill_position(B, create_coordinate(1, 1), 8)
B = board_fill_position(B, create_coordinate(1, 2), 2)
B = board_fill_position(B, create_coordinate(1, 3), 4)
B = board_fill_position(B, create_coordinate(2, 1), 8)
B = board_fill_position(B, create_coordinate(2, 2), 2)
B = board_fill_position(B, create_coordinate(2, 4), 4)
B = board_fill_position(B, create_coordinate(3, 1), 8)
B = board_fill_position(B, create_coordinate(3, 3), 4)
B = board_fill_position(B, create_coordinate(3, 4), 2)
B = board_fill_position(B, create_coordinate(4, 1), 8)
B = board_fill_position(B, create_coordinate(4, 3), 2)
B = board_fill_position(B, create_coordinate(4, 4), 4)
B = board_reduce(B, 'W')
print_board(B)
B = create_board()
B = board_fill_position(B, create_coordinate(1, 1), 16)
B = board_fill_position(B, create_coordinate(1, 2), 8)
B = board_fill_position(B, create_coordinate(1, 3), 4)
B = board_fill_position(B, create_coordinate(1, 4), 2)
B = board_reduce(B, 'N')
B = board_reduce(B, 'S')
B = board_reduce(B, 'E')
B = board_reduce(B, 'W')
print_board(B)