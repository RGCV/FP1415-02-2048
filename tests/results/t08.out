0
4
[ 2 ] [ 0 ] [ 0 ] [ 0 ] 
[ 2 ] [ 0 ] [ 0 ] [ 0 ] 
[ 2 ] [ 0 ] [ 0 ] [ 0 ] 
[ 2 ] [ 0 ] [ 0 ] [ 0 ] 
Score: 0
[ 0 ] [ 0 ] [ 0 ] [ 4 ] 
[ 0 ] [ 0 ] [ 0 ] [ 4 ] 
[ 0 ] [ 0 ] [ 0 ] [ 4 ] 
[ 0 ] [ 0 ] [ 0 ] [ 4 ] 
Score: 16
[ 4 ] [ 0 ] [ 0 ] [ 0 ] 
[ 4 ] [ 0 ] [ 0 ] [ 0 ] 
[ 4 ] [ 2 ] [ 0 ] [ 0 ] 
[ 4 ] [ 2 ] [ 0 ] [ 0 ] 
Score: 16
[ 4 ] [ 4 ] [ 4 ] [ 4 ] 
[ 2 ] [ 2 ] [ 4 ] [ 4 ] 
[ 0 ] [ 0 ] [ 0 ] [ 0 ] 
[ 0 ] [ 0 ] [ 0 ] [ 0 ] 
Score: 16
[ 0 ] [ 0 ] [ 0 ] [ 0 ] 
[ 0 ] [ 0 ] [ 0 ] [ 0 ] 
[ 4 ] [ 4 ] [ 4 ] [ 4 ] 
[ 4 ] [ 4 ] [ 4 ] [ 4 ] 
Score: 16
[ 4 ] [ 4 ] [ 0 ] [ 0 ] 
[ 4 ] [ 4 ] [ 0 ] [ 0 ] 
[ 4 ] [ 2 ] [ 0 ] [ 0 ] 
[ 4 ] [ 2 ] [ 0 ] [ 0 ] 
Score: 8
[ 0 ] [ 0 ] [ 2 ] [ 4 ] 
[ 0 ] [ 0 ] [ 2 ] [ 4 ] 
[ 0 ] [ 0 ] [ 2 ] [ 4 ] 
[ 0 ] [ 0 ] [ 4 ] [ 2 ] 
Score: 0
[ 2 ] [ 2 ] [ 4 ] [ 2 ] 
[ 4 ] [ 4 ] [ 2 ] [ 4 ] 
[ 0 ] [ 0 ] [ 0 ] [ 0 ] 
[ 0 ] [ 0 ] [ 0 ] [ 0 ] 
Score: 0
[ 0 ] [ 0 ] [ 0 ] [ 0 ] 
[ 0 ] [ 0 ] [ 2 ] [ 0 ] 
[ 4 ] [ 4 ] [ 4 ] [ 2 ] 
[ 2 ] [ 2 ] [ 8 ] [ 16 ] 
Score: 16
[ 8 ] [ 2 ] [ 4 ] [ 0 ] 
[ 8 ] [ 2 ] [ 4 ] [ 0 ] 
[ 8 ] [ 4 ] [ 2 ] [ 0 ] 
[ 8 ] [ 2 ] [ 4 ] [ 0 ] 
Score: 0
[ 0 ] [ 0 ] [ 0 ] [ 0 ] 
[ 0 ] [ 0 ] [ 0 ] [ 0 ] 
[ 0 ] [ 0 ] [ 0 ] [ 0 ] 
[ 16 ] [ 8 ] [ 4 ] [ 2 ] 
Score: 0
